### PR TITLE
[android] Shrink the APK by compressing assets and inflating them on load

### PR DESCRIPTION
**Pull request by biodranik**
Reduce .apk size by ~8Mb by compressing assets that can be safely uncompressed when loaded.

Pre-requisite to store SVG icons in the apk.

### CODE REVIEW

**Review by biodranik — COMMENTED** [review]
This is an experimental LLM code review

Overall this is a well-crafted PR. Logic is correct, the edge-case test (equal compressed/uncompressed sizes guarded by compression method, not size comparison) is exactly the right thing to test, and all `.sdf`→`.xml` references have been updated consistently across code, docs, tools, and Xcode projects. No correctness bugs found in the primary logic. A few smaller observations:

---

**[Medium] `CreateModelReader` opens the zip container twice for every deflated entry**

For a compressed asset the constructor builds a `ZipFileReader` (opening the APK to read the local-file header and find the data offset), then immediately resets it and calls `UnzipFile` which re-opens the same container to inflate. The two-open round-trip is harmless correctness-wise (Android's APK is backed by the OS page cache), but the metadata fetch in the first pass is then discarded.

A cleaner path would be to expose an inflate-from-known-offset helper so the same open can deliver both the header info and the inflated bytes. Alternatively, a simpler fix: let `UnzipFile` accept optional pre-resolved offset/size parameters that `CreateModelReader` can pass in after the first open.

---

**[Low] `ASSERT` instead of `CHECK` for bounds in `MemModelReader::Read` / `CreateSubReader`**

```cpp
ASSERT_LESS_OR_EQUAL(pos + size, m_size, (pos, size));
```

`ASSERT` is stripped in release builds. Any caller passing an out-of-bounds `pos`/`size` silently causes a buffer over-read in release. Since `MemModelReader` is a new public-facing `ModelReader` implementation (returned from `CreateModelReader`), `CHECK_LESS_OR_EQUAL` would be safer and consistent with the project's convention of using `CHECK` for always-on invariants. See CLAUDE.md: *"ASSERT is debug-only, CHECK is always-on."*

---

**[Low] `std::bad_alloc` during inflation bypasses the `Reader::OpenException` handler**

In `platform_android.cpp`:

```cpp
try
{
    return ZipFileReader::CreateModelReader(...);
}
catch (Reader::OpenException const & e)
{
    LOG(LWARNING, ...);
}
```

`MemBufferDelegate::OnBlockUnzipped` calls `m_buffer.insert(...)` which can throw `std::bad_alloc` if the device is low on memory. That exception is not a `Reader::OpenException`, so it propagates uncaught up the call stack and terminates the process rather than logging a warning and falling through to `FileAbsentException`. On low-end Android devices with 2 GB RAM, inflating `drules_proto_*.bin` (1–3 MB) plus other compressed assets around startup could trigger this. Adding a broader `catch (std::exception const & e)` or wrapping the inflation buffer with a `try`/`catch (std::bad_alloc)` inside `CreateModelReader` would make the failure path graceful.


---
_Generated by [Claude Code](https://claude.ai/code)_

File: libs/coding/zip_reader.cpp [hunk 1]
```diff
@@ -39,12 +46,56 @@ class UnzipFileDelegate : public ZipFileReader::Delegate
...
+    : ModelReader(name)
+    , m_buffer(std::move(buffer))
+    , m_offset(offset)
+    , m_size(size)
+  {}
+
+  std::shared_ptr<std::vector<char>> m_buffer;
+  size_t m_offset;
+  size_t m_size;
+};
```
Comment (vng) [inline libs/coding/zip_reader.cpp:91]: Avoid copy-paste. Use (with minor adaptation) BufferReader

**Review by strider-dunadan — APPROVED** [review]

**Review by vng — APPROVED** [review]
TODO: Unify with BufferReader.
